### PR TITLE
Copy sample to /endpoints and make it generic.

### PR DESCRIPTION
**Pull request by sepehre**
We plan on cleaning up our docs for the various platforms and share this
sample throughout the various quickstart guides.

It no longer makes sense to have this under appgengine/flexible. Instead
it will live under endpoints/getting-started. The appengine/flexible
copy will be deleted after the cloud site docs have been updated and
released.

This also introduces how to deploy this sample using
GAE/GKE/GCE/GCE+Docker.

### CODE REVIEW

File: endpoints/getting-started/Dockerfile.custom [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    apt-get install -y build-essential curl php5 php5-cli && \
+    apt-get clean && \
+    rm /var/lib/apt/lists/*_*
+# Install Composer
+RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin/ --filename=composer
+RUN chmod 777 /usr/local/bin/composer
+
+COPY * /app/
+RUN cd /app/ && /usr/local/bin/composer install
+
```
Comment (jeffmendoza) [inline endpoints/getting-started/Dockerfile.custom]: You need `WORKDIR /app` or else the php command does not run in the /app directory.

File: endpoints/getting-started/Dockerfile.custom [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ENTRYPOINT ["/usr/bin/php", "-S", "localhost:8081"]
```
Comment (jeffmendoza) [inline endpoints/getting-started/Dockerfile.custom]: Change to `CMD ["/usr/bin/php", "-S", "0.0.0.0:8081"]` Entrypoint overrides alternative docker commands. Using `localhost` will not let the container accept connections from any other IP.

File: endpoints/getting-started/Dockerfile.custom (hunk 1 above)
Comment (sepehre) [inline endpoints/getting-started/Dockerfile.custom]: done

**Comment by jeffmendoza** [comment]
Looks good

@bshaffer please merge when ready.

**Comment by bshaffer** [comment]
I'm not a fan of the directory name `getting-started`. The documentation uses `quickstart`, and the our other samples use `app`. We should use one of these instead. I do not believe `getting-started` is used in the other languages either.

**Comment by sepehre** [comment]
1. The name was chosen consistently with other language specific repos.
https://github.com/GoogleCloudPlatform/ruby-docs-samples/tree/master/endpoints/getting-started
https://github.com/GoogleCloudPlatform/golang-samples/tree/master/endpoints/getting-started
https://github.com/GoogleCloudPlatform/nodejs-docs-samples/tree/master/endpoints/getting-started

Other samples will be added under /endpoints/ in the future.

2. We are removing the old appengine/flexible samples until the
documentation is updated. A second round of PRs will be sent to remove the
old samples.

**Comment by bshaffer** [comment]
We should include a `README.md` in `appengine/flexible/endpoints` to point to this directory and the URL for the tutorial.

**Comment by sepehre** [comment]
appengine/flexible/endpoints already has a README.md with links to the
cloud site documentation.

And as I mentioned, appengine/flexible/endpoints directory will be deleted
shortly once our cloud site documentation has been updated.
This sample will be referenced from these quickstarts:
https://cloud.google.com/endpoints/docs/quickstarts not new tutorials.

**Comment by jeffmendoza** [comment]
@bshaffer do you have any other concerns? @sepehre covered the reasoning behind the copy, update docs, then delete plan, and the naming for the directory.

**Comment by bshaffer** [comment]
Yeah it seems good, I would like to see some E2E tests also, but we can add those later

**Comment by jeffmendoza** [comment]
I added myself to #193

**Comment by bshaffer** [comment]
I've added e2e tests and updated for `env:flex` in #243

**Pull request closed without merging**